### PR TITLE
Fix snowflake stage not being deleted on error in load_file

### PR DESCRIPTION
**Pull request by feluelle**
# Description

## What is the current behavior?

Currently the snowflake stage is not being dropped incase of an exception when calling `load_file`.

closes: #1262

## What is the new behavior?

Drop snowflake stage also when file load failed.

## Does this introduce a breaking change?

No.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sunank200 — APPROVED** [review]
LGTM

**Review by tatiana — APPROVED** [review]

**Review by kaxil — COMMENTED** [review]
One of the test is failing - https://github.com/astronomer/astro-sdk/actions/runs/3572950837/jobs/6007534322

File: python-sdk/src/astro/databases/snowflake.py [hunk 1]
```diff
@@ -615,13 +615,13 @@ def load_file_to_table_natively(
         except AttributeError:
             try:
                 rows = self.hook.run(sql_statement)
-            except (AttributeError, ValueError) as exe:
```
[the remark this answers is not in the document]
Comment (feluelle) [reply python-sdk/src/astro/databases/snowflake.py:618]: @kaxil @utkarsharma2 @sunank200 maybe?

[the remark this answers is not in the document]
Comment (sunank200) [reply python-sdk/src/astro/databases/snowflake.py:618]: It's required as this is raised by snowflake for which we should fallback if enabled by the user. Related to https://github.com/astronomer/astro-sdk/pull/1324

**Review by sunank200 — CHANGES_REQUESTED** [review]
`AttributeError` would be required as this is raised by snowflake for which we should fallback if enabled by the user. Related to #1324

File: python-sdk/src/astro/databases/snowflake.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sunank200) [reply python-sdk/src/astro/databases/snowflake.py:618]: Tested it now its working without `AttributeError`

**Review by sunank200 — APPROVED** [review]

**Comment by sunank200** [comment]
> `AttributeError` would be required as this is raised by snowflake for which we should fallback if enabled by the user. Related to #1324

Tested it without `AttributeError` and it works so the change is fine

**Pull request merged**